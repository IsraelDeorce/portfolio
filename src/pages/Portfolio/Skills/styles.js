import styled from 'styled-components'

export const Skills = styled.section``

export const Progress = styled.div`
  display: flex;
  align-self: center;
  flex-direction: column;
  max-width: 600px;
  & > *:not(:last-child) {
    margin-bottom: 1em;
  }
`
