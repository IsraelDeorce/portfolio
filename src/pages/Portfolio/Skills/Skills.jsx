import { useTranslation } from 'react-i18next'
import { ProgressIndicator } from '../../../components'
import * as S from './styles'

function Skills() {
  const { t } = useTranslation()

  return (
    <S.Skills id='skills'>
      <h2>{t('button.skills')}</h2>
      <S.Progress>
        <ProgressIndicator percentage={99} description={t('skills.software_engineering')} />
        <ProgressIndicator
          percentage={90}
          description={`${t('skills.front')} (${t('skills.react')})`}
        />
        <ProgressIndicator
          percentage={50}
          description={`${t('skills.back')} (${t('skills.node')}, ${t('skills.dotnet')})`}
        />
        <ProgressIndicator percentage={25} description={t('skills.db')} />
        <ProgressIndicator percentage={50} description={t('skills.devops')} />
        <ProgressIndicator percentage={90} description={t('skills.automated_tests')} />
      </S.Progress>
    </S.Skills>
  )
}

export default Skills
