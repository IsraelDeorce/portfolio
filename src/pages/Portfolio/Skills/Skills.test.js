import Skills from './Skills'

describe('<Skills/>', () => {
  describe('Snapshot', () => {
    it('renders without crashing using shallow', () => {
      expect(shallow(<Skills />)).toMatchSnapshot()
    })
  })
})
