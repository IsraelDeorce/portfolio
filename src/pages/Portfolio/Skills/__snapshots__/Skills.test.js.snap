// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skills/> Snapshot renders without crashing using shallow 1`] = `
<section
  id="skills"
>
  <h2>
    button.skills
  </h2>
  <styled.div>
    <ProgressIndicator
      description="skills.software_engineering"
      percentage={99}
    />
    <ProgressIndicator
      description="skills.front (skills.react)"
      percentage={90}
    />
    <ProgressIndicator
      description="skills.back (skills.node, skills.dotnet)"
      percentage={50}
    />
    <ProgressIndicator
      description="skills.db"
      percentage={25}
    />
    <ProgressIndicator
      description="skills.devops"
      percentage={50}
    />
    <ProgressIndicator
      description="skills.automated_tests"
      percentage={90}
    />
  </styled.div>
</section>
`;
